File: pi4/c/examples/1.2.0_Button/button.c
/*
 This example works with the Mosiwi Basic learning board.

 Learning target:
	1. Button
	2. Variable: char, global variable
	3. Assignment operator: =
	4. Digital pin input Settings
	5. Relational operator: ==, !=
	6. Judgment statement: if

 Web: http://mosiwi.com
 Wiki: https://mosiwi-wiki.readthedocs.io
 Designer: jalen
 Date：2023-5-16
*/
#include<wiringPi.h>

// Global variable.
char buttonPin = 22;   // the number of the pushbutton pin
char ledPin = 23;      // the number of the LED pin
char buttonState = 0;  // variable for reading the pushbutton status


int main(void) {
	wiringPiSetup();
	pinMode(ledPin, OUTPUT);
	pinMode(buttonPin, INPUT);

	while(1){
		// read the state of the pushbutton value:
		buttonState = digitalRead(buttonPin);

		// check if the pushbutton is pressed. If it is, the buttonState is LOW:
		if (buttonState == LOW) {
		// turn LED on:
		digitalWrite(ledPin, HIGH);
		}
		if (buttonState != LOW){
		// turn LED off:
		digitalWrite(ledPin, LOW);
		}
	}
}

File: pico/microPython/button.py
import time                  # Importing a time class
from machine import Pin      # Import the Pin class from the machine module.

Button = Pin(21, Pin.IN)     # create input pin on GPIO21
G_LED = Pin(9, Pin.OUT)      # create output pin on GPIO9

while(1):                    # An infinite loop statement.
    button_value = Button.value()  # Read the key value
    if button_value == 1:          # Check whether the value of the key is 1, if so, turn off the LED.
        G_LED.off()                # Set pin to "off" (low) level
    else:                          # If the value of the key is not 1, the LED is turned on.
        G_LED.on()                 # Set pin to "on" (high) level